README.md

```diff
@@ -745,12 +745,12 @@ _You can enable the following settings in Xcode by running [this script](resourc
     func request(completion: () -> Void) {
       API.request() { [weak self] response in
         guard let self = self else { return }
-        self.doSomething(self.property)
+        self.doSomething(with: self.property, response: response)
         completion()
       }
     }
 
-    func doSomething(nonOptionalParameter: SomeClass) {
+    func doSomething(with nonOptionalParameter: SomeClass, response: SomeResponseClass) {
```

Comment:
Could we give the first parameter a more descriptive name?
```suggestion
    func doSomething(withNonOptionalParameter nonOptionalParameter: SomeClass, response: SomeResponseClass) {
```

Open to suggestions, but `with` isn't terrible descriptive as-is.

Reply:
Yeah I considered this. Generally the class type would give you the context rather than a wordy label, so in reality `with` I think would be okay. Hard to be descriptive yet not overly verbose when your property is called `property` and the class is called `SomeClass`

Reply:
Looks like [Swift's API guidelines](https://swift.org/documentation/api-design-guidelines/#promote-clear-usage) overrule me here. I got tripped up by our [hint at types](https://github.com/airbnb/swift#hint-at-types) rule, but our guide [does not override Swift's guidelines](https://github.com/airbnb/swift#guiding-tenets).

Reply:
I actually tend to disagree. 

Based on [Promote Clear Usage](https://swift.org/documentation/api-design-guidelines/#promote-clear-usage )

```swift
public mutating func remove(at position: Index) -> Element
```

and [Parameter names](https://swift.org/documentation/api-design-guidelines/#parameter-names)

```swift
func move(from start: Point, to end: Point)
```

it's clear to me that we don't need to add more information to the parameter name when the type is strong.

This is not true when the type information is weak (e.g. a String or an Int). In that case it's better to move this information to the function name directly and then in the subsequent parameters.

See **Compensate for weak type information**

```diff
- func add(_ observer: NSObject, for keyPath: String)
+ func addObserver(_ observer: NSObject, forKeyPath path: String)
```